File: js/leetcode/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  let chars = s.replace(/[\W_]+/g, "").toLowerCase().split('');
  let left = 0; right = chars.length - 1;
  while (left <= right) {
    if (chars[left] != chars[right]) {
      return false;
    }
    else {
      left++;
      right--;
    }
  }
  return true;
};

File: python/leetcode/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        sentence = ''.join(ch for ch in s if ch.isalnum()).lower()
        return sentence == sentence[::-1]


class Solution:
    def isPalindrome(self, s: str) -> bool:
        chars = [ch for ch in s.lower() if ch.isalnum()]

        left, right = 0, len(chars)-1

        while left < right:
            if chars[left] != chars[right]:
                return False
            left += 1
            right -= 1

        return True
